src/charm.py

```diff
@@ -136,10 +153,10 @@ def _process_wheel_file(self, event):  # noqa: C901
                 wheel_data = file.read()
 
                 # Push wheel file to the container and extract it.
-                container.push("/user_provided/wheel_file.whl", wheel_data, make_dirs=True)
+                container.push(f"/user_provided/{filename}", wheel_data, make_dirs=True)
                 container.exec(["apt-get", "update"]).wait()
                 container.exec(["apt-get", "install", "unzip"]).wait()
-                container.exec(["unzip", "/user_provided/wheel_file.whl", "-d", "/user_provided"]).wait()
+                container.exec(["unzip", f"/user_provided/{filename}", "-d", "/user_provided"]).wait()
```

Comment:
If you are going to reuse paths it might be better to put it in a variable.

Later Edit: It looks like you reuse this a lot below.